wordpress-mobile/WordPress-Android
# Layout Grid Availability -  featureConfig implementation

This PR is similar to https://github.com/wordpress-mobile/WordPress-Android/pull/13804

It adds the availability information that is needed for the layout grid block to be available to only .com sites for now and in the future to all sites that have the layout grid block.

Related PR:
https://github.com/WordPress/gutenberg/pull/31777
Related https://github.com/wordpress-mobile/gutenberg-mobile/pull/3513

To test:

CASE A: 
    - An business site as well as a simple site should see the layout grid block.

CASE B: 
     - An Jetpack site or a .org site should NOT see the layout grid block.
